packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -425,10 +425,19 @@ export class Cluster extends ClusterBase {
     }
 
     const clusterType = props.clusterType || ClusterType.MULTI_NODE;
-    const nodeCount = props.numberOfNodes !== undefined ? props.numberOfNodes : (clusterType === ClusterType.MULTI_NODE ? 2 : 1);
 
-    if (clusterType === ClusterType.MULTI_NODE && nodeCount < 2) {
-      throw new Error('Number of nodes for cluster type multi-node must be at least 2');
+    let numberOfNodes;
+
+    if (clusterType === ClusterType.MULTI_NODE) {
+      if (props.numberOfNodes ===  undefined || props.numberOfNodes < 2) {
+        throw new Error('Number of nodes for cluster type multi-node must be at least 2');
+      }
+      numberOfNodes = props.numberOfNodes;
+    } else {
+      // SINGLE_NODE
+      if (props.numberOfNodes !==  undefined) {
+        throw new Error('Number of nodes for cluster type single-node must not be set2');
+      }
```

Comment:
Would be nice to explicitly set this to 1, just to be clear

```suggestion
      }
      numberOfNodes = 1;
```

Reply:
I've amended my code reflecting your suggestions.

What is the purpose of setting numberOfNodes to 1? 

As per the documentation (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-redshift-cluster.html#cfn-redshift-cluster-clustertype), ClusterType is required and if single-node, the NumberOfNodes parameter is not required, if multi-node, the NumberOfNodes parameter is required. When creating a CF stack with single-node cluster and 1 node explicitly set, my generated CF templated was rejected - see my original error message #8887

Reply:
In that case, ignore this comment.